src/CardDav/Backend.php

```diff
@@ -178,61 +167,43 @@ public function getVcards(): array
-        if (strcasecmp($serv[2], $link[2]) !== 0) {            // if they aren´t equal authorisation will fail!
-            return $vcard;
-        }
-
-        $embedded = $this->getLinkedData($vcard->{$substituteID});   // get the data from the external URL
-        $types = '';
-        switch ($vcard->version) {                             // the different vCard versions must be considered
-            case '3.0':
-                $types = "TYPE=" . strtoupper($embedded['subtype']) . ";ENCODING=b";
-                break;
-            case '4.0':
-                $types = "data:" . $embedded['mimetype'] . ";base64";
-                break;
+        if ($embedded = $this->getLinkedData($vcard->$property)) {      // get the data from the external URL or false
+            if ($vcard->VERSION == '3.0') {                             // the different vCard versions must be considered
+                unset($vcard->$property);                               // delete the old property
+                $vcard->add($property, $embedded['data'], ['TYPE' => strtoupper($embedded['subtype']), 'ENCODING' => 'b']);
+            } elseif ($vcard->VERSION == '4.0') {
+                unset($vcard->$property);                               // delete the old property
+                $vcard->add($property, 'data:' . $embedded['mimetype'] . ';base64,' . base64_encode($embedded['data']));
+            }
         }
 
-        $rawField  = "raw" . ucfirst($substituteID);
-        $dataField = $substituteID . "Data";
-        $vcard->$rawField  = $embedded['data'];
-        $vcard->$dataField = $types;
-
         return $vcard;
     }
 
     /**
      * Delivers an array including the previously linked data and its mime type details
      * a mime type is composed of a type, a subtype, and optional parameters (e.g. "; charset=UTF-8")
      *
-     * @param    string $uri           URL of the external linked data
-     * @return   array ['mimetype',    e.g. "image/jpeg"
-     *                  'type',        e.g. "audio"
-     *                  'subtype',     e.g. "mpeg"
-     *                  'parameters',  whatever
-     *                  'data']        the base64 encoded data
-     * @throws \Exception
+     * @param string $uri           URL of the external linked data
+     * @return bool|array ['mimetype',    e.g. "image/jpeg"
+     *                     'type',        e.g. "audio"
+     *                     'subtype',     e.g. "mpeg"
+     *                     'parameters',  whatever
+     *                     'data']        the base64 encoded data
      */
-    public function getLinkedData(string $uri): array
+    private function getLinkedData(string $uri)
     {
-        $response = $this->getCachedClient()->request('GET', $uri);
+        $response = $this->getCachedClient()->request('GET', $uri, ['http_errors' => false]);
+        if ($response->getStatusCode() != 200) {
```

Comment:
Client should throw an exception if not 200 so this should never be reached?

Reply:
To cover #122:
if download linked data failed (statuscode <> 200) now no exception will thrown (['http_errors' => false]). This corresponds to line 104 - 108 in functions.php.
I prefer a message there instead of during download vcards: easier to read than inside progress